zeek/README.md

```diff
@@ -1,6 +1,412 @@
+- [Customizing type definitions](#customizing-type-definitions)
+  * [Sample data](#sample-data-1)
+  * [Handling exceptions](#handling-exceptions)
+- [Older versions of Zeek/Bro](#older-versions-of-zeekbro)
+- [Need help? Have feedback?](#need-help-have-feedback)
 
-In the future it is expected to contain additional versions, generated on different versions/configurations of Zeek.
+# Summary
 
+The file [`types.json`](types.json) in this directory contains configuration
+that can be used to restore Zeek-compatible rich data types to events that
+have been output in JSON format by Zeek. The definitions in the current
+revision of `types.json` reflect the default set of logs output by
+[Zeek v3.1.2](https://github.com/zeek/zeek/releases/tag/v3.1.2) and are
+likely to be compatible with default logs from older Zeek/Bro releases as
+well.
+
+If your Zeek environment is customized, you will need to adjust the type
+definition to match the differences in your JSON logs. The sections below
+describe the customization process.
+
+# Contact us!
+
+If you're using JSON type defintions with `zq` or [Brim](https://github.com/brimsec/brim),
+we'd like to hear from you! We know the process of customizing the definitions
+can be tricky. We have ideas for ways we might improve it, but we'll have a
+better sense of priority and how to go about it if we've heard from those
+who have tried it. Whether you've used this feature and it "just worked" or if
+you hit challenges and need help, please join our
+[public Slack](https://join.slack.com/t/brimsec/shared_invite/zt-cy34xoxg-hZiTKUT~1KdGjlaBIuUUdg)
+and tell us about it. Thanks!
+
+# Sample data
+
+To see a working example before working with your own data, clone the
+[zq-sample-data](https://github.com/brimsec/zq-sample-data) repository.
+
+```
+# git clone --depth=1 https://github.com/brimsec/zq-sample-data ~/zq-sample-data
+```
+
+A set of JSON-format Zeek logs will now be present in `~/zq-sample-data/zeek-ndjson`.
+
+# Usage
+
+Asssuming you have the [`zq`](https://github.com/brimsec/zq) repository locally
+cloned to `~/zq` and the `zq` binary is in your `$PATH`, here's an example of
+reading in all Zeek events while applying the JSON type definition:
+
+```
+# zq -f table -j ~/zq/zeek/types.json "count()" ~/zq-sample-data/zeek-ndjson/*
+COUNT
+1462078
+```
+
+Since we saw no errors from `zq`, that means the definitons in `types.json`
+fully described all named fields in all the log files. If there had been
+errors, `zq` would have stopped reading when the first error was encountered
+and would not have returned a `count()` result. The sections below describe
+what to do if we'd seen such errors here.
+
+# Why is this even necessary?
+
+Consider this Zeek HTTP event as output by the
+[JSON Streaming Logs](https://github.com/corelight/json-streaming-logs)
+package:
+
+```
+# gzcat ~/zq-sample-data/zeek-ndjson/http.ndjson.gz | head -1 | jq -S .
```

Comment:
nit: isn't `head -n 1` more usual?

Reply:
Hunh. I use `head` so rarely that I'd never noticed that. I just did some reading and saw that it's shorthand that's undocumented in many implementations. So, yeah... changing it to `-n 1`.